src/server/coprocessor/endpoint.rs

```diff
@@ -115,20 +117,40 @@ pub struct RequestTask {
     deadline: Instant,
     statistics: Statistics,
     on_resp: OnResponse,
+    sel_req: Option<SelectRequest>,
```

Comment:
we must have a select req or an err, so why not using Option<Result> here directly?